src/dotnet/CommandLine/CommandLineApplication.cs

```diff
@@ -334,7 +340,15 @@ public void ShowHelp(string commandName = null)
             for (var cmd = this; cmd != null; cmd = cmd.Parent)
             {
                 cmd.IsShowingInformation = true;
-                headerBuilder.Insert(6, string.Format(" {0}", cmd.Name));
+                if (cmd != this && cmd.Arguments.Any())
+                {
+                    var args = string.Join(" ", cmd.Arguments.Select(arg => arg.Name));
+                    headerBuilder.Insert(6, string.Format(" {0} {1}", cmd.Name, args));
```

Comment:
should this join all the parents' names too before cmd.Name? I'm not sure what the cmd.Parent is in this context but if it's what the name suggest it seems to me that we will get repetition and invalid command invokation in some cases